core/src/main/java/com/scalar/db/storage/jdbc/RdbEngineOracle.java

```diff
@@ -202,6 +206,82 @@ public String getDataTypeForKey(DataType dataType) {
     }
   }
 
+  @Override
+  public DataType getDataTypeForScalarDb(
+      JDBCType type, String typeName, int columnSize, int digits, String columnDescription) {
+    String numericTypeDescription = String.format("%s(%d, %d)", typeName, columnSize, digits);
+    switch (type) {
+      case NUMERIC:
+        if (columnSize > 15) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "data type %s is unsupported: %s", numericTypeDescription, columnDescription));
+        }
+        if (digits == 0) {
+          logger.info(
+              "data type larger than that of underlying database is assigned: {} to BIGINT",
+              numericTypeDescription);
+          return DataType.BIGINT;
+        } else {
+          logger.info(
+              "fixed-point data type is casted, be aware round-up or round-off can be happen in underlying database: {} ({} to DOUBLE)",
+              columnDescription,
+              numericTypeDescription);
+          return DataType.DOUBLE;
+        }
+      case REAL:
+        return DataType.FLOAT;
+      case FLOAT:
+        if (columnSize > 53) {
+          throw new IllegalArgumentException(
+              String.format(
+                  "data type %s is unsupported: %s", numericTypeDescription, columnDescription));
+        }
+        if (columnSize < 53) {
+          logger.info(
+              "data type larger than that of underlying database is assigned: {} to DOUBLE",
+              numericTypeDescription);
+        }
+        return DataType.DOUBLE;
+      case DOUBLE:
+        logger.warn(
+            String.format(
+                "data type that may be smaller than that of underlying database is assigned: %s (Oracle %s to our BIGINT)",
+                columnDescription, typeName));
+        return DataType.DOUBLE;
+      case CHAR:
+      case NCHAR:
+      case VARCHAR:
+      case NVARCHAR:
+      case CLOB:
+      case NCLOB:
+        logger.info(
+            "data type larger than that of underlying database is assigned: {} ({} to TEXT)",
+            columnDescription,
+            typeName);
+        return DataType.TEXT;
+      case LONGVARCHAR:
+        return DataType.TEXT;
+      case VARBINARY:
+        logger.info(
+            "data type larger than that of underlying database is assigned: {} (Oracle {} to BLOB)",
+            columnDescription,
+            typeName);
+        return DataType.BLOB;
+      case LONGVARBINARY:
+        return DataType.BLOB;
+      case BLOB:
+        logger.info(
```

Comment:
This should be `warn`?

Reply:
I guess this should be `info`, but the message was wrong. Anyway, good catch! Thank you!

Reply:
Let me take that back. I think I used the "smaller" message since the maximum size is larger than the logical maximum size of ScalarDB BLOB. So, this should be `warn` as you noticed.

https://docs.oracle.com/en/database/oracle/oracle-database/21/sqlrf/Data-Types.html#GUID-7B72E154-677A-4342-A1EA-C74C1EA928E6

> A binary large object. Maximum size is (4 gigabytes - 1) * (database block size).

Reply:
Fixed in 0d3a2f8.